File: dp-ui/src/api/messageChannel.ts
import request from './request'
import type { ApiResponse } from '@/types/api'

export interface MessageChannel {
  id: number
  channelName: string
  channelType: 'email' | 'wecom' | 'dingtalk' | 'sms'
  config: string
  isDefault: number
  status: number
  description?: string
  createBy?: number
  createTime?: string
  updateTime?: string
}

export interface EmailConfig {
  host: string
  port: number
  username: string
  password: string
  fromName?: string
  ssl?: boolean
}

export interface WecomConfig {
  corpId: string
  agentId: string
  secret: string
  webhookUrl?: string
}

export interface DingtalkConfig {
  webhookUrl: string
  secret?: string
  agentId?: string
  appKey?: string
  appSecret?: string
}

export interface SmsConfig {
  provider: 'aliyun' | 'tencent'
  accessKey: string
  secretKey: string
  signName: string
  templateId?: string
}

/**
 * 获取所有通道配置
 */
export const getMessageChannels = () => {
  return request.get<ApiResponse<MessageChannel[]>>('/message-channel/list')
}

/**
 * 按类型获取通道配置
 */
export const getChannelsByType = (type: string) => {
  return request.get<ApiResponse<MessageChannel[]>>(`/message-channel/by-type/${type}`)
}

/**
 * 获取启用的通道（按类型分组）
 */
export const getEnabledChannels = () => {
  return request.get<ApiResponse<Record<string, MessageChannel[]>>>('/message-channel/enabled')
}

/**
 * 获取单个通道详情
 */
export const getChannelById = (id: number) => {
  return request.get<ApiResponse<MessageChannel>>(`/message-channel/${id}`)
}

/**
 * 创建通道配置
 */
export const createChannel = (data: Partial<MessageChannel>) => {
  return request.post<ApiResponse<MessageChannel>>('/message-channel', data)
}

/**
 * 更新通道配置
 */
export const updateChannel = (id: number, data: Partial<MessageChannel>) => {
  return request.put<ApiResponse<MessageChannel>>(`/message-channel/${id}`, data)
}

/**
 * 删除通道配置
 */
export const deleteChannel = (id: number) => {
  return request.delete<ApiResponse<void>>(`/message-channel/${id}`)
}

/**
 * 设置为默认通道
 */
export const setDefaultChannel = (id: number) => {
  return request.post<ApiResponse<void>>(`/message-channel/${id}/set-default`)
}

/**
 * 测试通道配置
 */
export const testChannel = (id: number, recipient: string, content?: string) => {
  return request.post<ApiResponse<string>>(`/message-channel/${id}/test`, { recipient, content })
}

File: dp-core/src/main/java/com/dataplatform/data/entity/MessageChannel.java
package com.dataplatform.data.entity;

import lombok.Data;
import java.util.Date;

/**
 * 消息通道配置实体
 * 统一管理企业微信、钉钉、邮件等消息通道配置
 * 每种类型可以配置多个，在使用时选择具体配置
 */
@Data
public class MessageChannel {
    private Long id;
    
    /** 通道名称（用于显示和选择） */
    private String channelName;
    
    /** 通道类型: email/wecom/dingtalk/sms */
    private String channelType;
    
    /** 配置内容（JSON格式，不同类型有不同配置项） */
    private String config;
    
    /** 是否默认配置：1-是，0-否（每种类型只能有一个默认） */
    private Integer isDefault;
    
    /** 是否启用：1-启用，0-禁用 */
    private Integer status;
    
    /** 描述说明 */
    private String description;
    
    /** 创建人 */
    private Long createBy;
    
    private Date createTime;
    private Date updateTime;
    
    // ==================== 各类型配置字段说明 ====================
    // email: {"host":"smtp.example.com","port":465,"username":"xxx","password":"xxx","fromName":"系统通知","ssl":true}
    // wecom: {"corpId":"xxx","agentId":"xxx","secret":"xxx","webhookUrl":"xxx(可选)"}
    // dingtalk: {"webhookUrl":"xxx","secret":"xxx(可选)","agentId":"xxx(可选)","appKey":"xxx(可选)","appSecret":"xxx(可选)"}
    // sms: {"provider":"aliyun/tencent","accessKey":"xxx","secretKey":"xxx","signName":"xxx","templateId":"xxx"}
}
